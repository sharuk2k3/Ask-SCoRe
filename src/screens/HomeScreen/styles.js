import {StyleSheet, Dimensions, Platform} from "react-native";
import {COLOR_LIGHT_PRIMARY, COLOR_LIGHT_SECONDARY} from "../../config/styles";

let deviceWidth = Dimensions.get("window").width;
let deviceHeight = Dimensions.get("window").height;

export default StyleSheet.create({
  HeaderStyle: {
    fontSize: 30,
    flex: 1,
    marginLeft: 10,
    marginTop: 5,
  },

  ImageView: {
    width: deviceWidth * 0.13,
    aspectRatio: 1,
    marginHorizontal: 5,
  },

  HeaderStyle1: {
    flexDirection: "row",
  },

  DividerView: {
    borderBottomColor: "black",
    borderBottomWidth: 1,
    marginBottom: 3,
    marginHorizontal: 10,
  },

  AskButtonStyle: {
    width: deviceWidth * 0.14,
    height: deviceHeight * 0.04,
    backgroundColor: COLOR_LIGHT_PRIMARY,
    borderRadius: 10,
    right: 10,
    justifyContent: "center",
    marginTop: Platform.OS == "ios" ? 7 : 12,
  },

  QuestionStyle: {
    width: "90%",
    borderRadius: 5,
    borderWidth: 1,
    backgroundColor: Platform.OS === "ios" ? "#fff" : null,
    borderColor: "#b8b8b8",
    shadowColor: "#000000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: Platform.OS === "ios" ? 0.9 : 0.8,
    shadowRadius: Platform.OS === "ios" ? 3 : 40,
    elevation: Platform.OS === "ios" ? 3 : 4,
    alignSelf: "center",
  },

  nameHeadlineStyle: {
    width: "101%",
    paddingVertical: 2,
    backgroundColor: COLOR_LIGHT_SECONDARY,
    borderRadius: 7,
    alignSelf: "center",
    justifyContent: "space-between",
    top: 0,
    flexDirection: "row",
  },

  AnswerButtonStyle: {
    width: "20%",
    paddingVertical: 2,
    backgroundColor: COLOR_LIGHT_PRIMARY,
    borderRadius: 5,
    alignSelf: "flex-end",
    bottom: 5,
    right: 5,
    justifyContent: "center",
  },

  AnswerButtonStyle1: {
    width: "20%",
    paddingVertical: 2,
    backgroundColor: COLOR_LIGHT_SECONDARY,
    borderRadius: 5,
    alignSelf: "flex-end",
    bottom: 5,
    right: 5,
    justifyContent: "center",
  },

  textStyle: {
    color: "black",
    fontWeight: "400",
    textAlign: "center",
    fontSize: 14,
  },

  AskStyle: {
    color: "black",
    fontWeight: "500",
    textAlign: "center",
    fontSize: 16,
  },

  itemnameStyle: {
    marginLeft: 10,
    marginTop: 2,
  },

  postTimeStyle: {
    fontSize: 10,
    marginTop: 3,
    marginRight: 5,
    color: "white",
  },

  QueryStyle: {
    padding: 10,
    fontSize: 14,
  },

  ImageStyle: {
    borderRadius: 10,
    height: deviceHeight * 0.14,
    aspectRatio: 1,
    alignSelf: "center",
  },
});
