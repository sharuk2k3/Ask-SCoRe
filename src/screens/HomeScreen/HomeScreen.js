import React from "react";
import {firebase} from "../../config/config";
import {
  View,
  Text,
  SafeAreaView,
  Image,
  TouchableOpacity,
  FlatList,
} from "react-native";
import {withNavigation} from "react-navigation";
import {LOGO} from "../../config/styles.js";
import styles from "./styles";

class HomeScreen extends React.Component {
  state = {queryDetail: []};
  componentDidMount() {
    const {queryDetail} = this.state;
    firebase
      .firestore()
      .collection("queries")
      .orderBy("postDateTime", "desc")
      .onSnapshot(querySnapshot => {
        var newEntity = [];
        var queryArray = [];
        querySnapshot.forEach(doc => {
          qData = {
            uid: doc.data().id,
            queryInput: doc.data().queryInput,
            postTime: doc.data().postTime,
            id: doc.id,
            postdDteTime: doc.data().postDateTime,
            answerPresent: doc.data().answerPresent,
            queryImage: doc.data().queryImage,
          };
          queryArray.push(qData);
        });

        queryArray.forEach(query => {
          firebase
            .firestore()
            .collection("users")
            .doc(query.uid)
            .get()
            .then(userObject => {
              data = {
                fullName: userObject.data().fullName,
                queryInput: query.queryInput,
                id: query.id,
                postTime: query.postTime,
                postDateTime: query.postDateTime,
                answerPresent: query.answerPresent,
                queryImage: query.queryImage,
              };
              newEntity.push(data);
              this.setState({queryDetail: newEntity});
            });
        });
      });
  }

  render() {
    const {queryDetail} = this.state;
    return (
      <SafeAreaView>
        <View>
          <View style={{marginBottom: 15}}>
            <View style={styles.HeaderStyle1}>
              <Image source={LOGO} style={styles.ImageView} />
              <Text style={styles.HeaderStyle}>Community</Text>
              <TouchableOpacity
                onPress={() => this.props.navigation.push("Question")}
                style={styles.AskButtonStyle}>
                <Text style={styles.AskStyle}>ASK</Text>
              </TouchableOpacity>
            </View>
            <View style={styles.DividerView} />
            <View style={styles.DividerView} />
          </View>

          <FlatList
            data={queryDetail}
            renderItem={({item}) => {
              return (
                <TouchableOpacity
                  onPress={() =>
                    this.props.navigation.push("Answer", {
                      data: item.id,
                      qData: item.queryInput,
                    })
                  }>
                  <View style={{marginVertical: 10}}>
                    <View style={styles.QuestionStyle}>
                      <View style={styles.nameHeadlineStyle}>
                        <Text style={styles.itemnameStyle}>
                          {item.fullName}
                        </Text>
                        <Text style={styles.postTimeStyle}>
                          {item.postTime}
                        </Text>
                      </View>
                      <View>
                        <Text style={styles.QueryStyle}>{item.queryInput}</Text>
                        {item.queryImage ? (
                          <Image
                            source={{uri: item.queryImage}}
                            style={styles.ImageStyle}
                          />
                        ) : null}
                        {item.answerPresent ? (
                          <View style={styles.AnswerButtonStyle1}>
                            <Text style={styles.textStyle}>Answered</Text>
                          </View>
                        ) : (
                          <View style={styles.AnswerButtonStyle}>
                            <Text style={styles.textStyle}>Answer</Text>
                          </View>
                        )}
                      </View>
                    </View>
                  </View>
                </TouchableOpacity>
              );
            }}
            keyExtractor={(item, index) => "key" + index}
            ListFooterComponent={<View style={{height: 170}} />}
          />
        </View>
      </SafeAreaView>
    );
  }
}

export default HomeScreen;
