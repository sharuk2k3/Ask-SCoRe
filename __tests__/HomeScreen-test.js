import React from "react";
import renderer from "react-test-renderer";
import HomeScreen from "../src/screens/HomeScreen/HomeScreen";

jest.useFakeTimers();

test("render HomeScreen Correctly", () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
