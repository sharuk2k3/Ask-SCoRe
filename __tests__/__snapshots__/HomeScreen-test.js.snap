// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render HomeScreen Correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source="test-file-stub"
          style={
            Object {
              "aspectRatio": 1,
              "marginHorizontal": 5,
              "width": 97.5,
            }
          }
        />
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 30,
              "marginLeft": 10,
              "marginTop": 5,
            }
          }
        >
          Community
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#9cd683",
              "borderRadius": 10,
              "height": 53.36,
              "justifyContent": "center",
              "marginTop": 7,
              "opacity": 1,
              "right": 10,
              "width": 105.00000000000001,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            ASK
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "marginBottom": 3,
            "marginHorizontal": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "marginBottom": 3,
            "marginHorizontal": 10,
          }
        }
      />
    </View>
    <RCTScrollView
      ListFooterComponent={
        <View
          style={
            Object {
              "height": 170,
            }
          }
        />
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "height": 170,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
